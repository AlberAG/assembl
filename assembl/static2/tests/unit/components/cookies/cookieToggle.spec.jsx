import React from 'react';
import renderer from 'react-test-renderer';
import CookieToggle from '../../../../js/app/components/cookies/cookieToggle';

describe('CookieToggle component', () => {
  const handleToggleSpy = jest.fn(() => {});
  it('should render a cookie\'s name with a toggle to change its setting', () => {
    const props = {
      cookie: {
        name: 'userSession',
        category: 'essential',
        accepted: true
      },
      handleToggle: handleToggleSpy
    };
    const rendered = renderer.create(<CookieToggle {...props} />);
    expect(rendered).toMatchSnapshot();
  });
});