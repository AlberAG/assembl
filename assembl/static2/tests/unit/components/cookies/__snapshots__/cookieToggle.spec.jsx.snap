// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieToggle component should render a cookie's name with a toggle to change its setting 1`] = `
<div
  className="cookie-toggle"
>
  <span
    className="cookie-title dark-title-3 ellipsis"
  >
    User Session
  </span>
  <div
    className="rsbc-switch-button rsbc-mode-switch rsbc-switch-button-flat-round disabled"
  >
    <label
      htmlFor="userSession"
    >
      Refuse
    </label>
    <input
      checked={false}
      disabled={true}
      id="userSession"
      name="userSession"
      onChange={[Function]}
      type="checkbox"
      value="1"
    />
    <label
      htmlFor="userSession"
    />
    <label
      htmlFor="userSession"
    >
      Accept
    </label>
  </div>
</div>
`;
